analyses/cell-type-wilms-tumor-06/scripts/06a_build-geneposition.R

```diff
@@ -1,25 +1,99 @@
-# This script downloads and adapt the genome position file for use with infercnv
+# load library
+library(tidyverse)
+
+# Setup ------------------
 
 # The base path for the OpenScPCA repository, found by its (hidden) .git directory
 repository_base <- rprojroot::find_root(rprojroot::is_git_root)
 # The path to this module
 module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
+# The path to the reference directory
+reference_dir <- file.path(module_base, "results", "references")
+
+# URLs for data to download
+gtf_file_uri <- "s3://scpca-references/homo_sapiens/ensembl-104/annotation/Homo_sapiens.GRCh38.104.gtf.gz"
+arm_order_file_url <- "http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/cytoBand.txt.gz"
 
-# load library
-library(tidyverse)
 # Define the gene order file
-  gene_order_file <- file.path(module_base, "results","references", 'gencode_v19_gene_pos.txt')
-
-# Download the file if ot existing
-if (!file.exists(gene_order_file)) {
-  download.file(url = 'https://data.broadinstitute.org/Trinity/CTAT/cnv/gencode_v19_gen_pos.complete.txt',
-                destfile = gene_order_file)
-  gene_order <- read_tsv(gene_order_file, col_names =  FALSE)
-  tmp <- gsub("\\..*","",gene_order$X1) 
-  tmp <- gsub(".*\\|","",tmp) 
-  gene_order$X1 <- tmp
-  gene_order <- gene_order[grepl("ENSG", x = gene_order$X1),]
-  
-  write_tsv(col_names = FALSE, gene_order, gene_order_file, append = FALSE)
+gtf_file <- file.path(reference_dir, "Homo_sapiens.GRCh38.104.gtf.gz")
+
+# Define the arm order file
+arm_order_file <- file.path(reference_dir, "hg38_cytoBand.txt")
+
+# Define the gene/arm combined information file for output
+gene_arm_order_file <- file.path(reference_dir, "gencode_v38_gene_pos_arm.txt")
+
+# Download files if they don't exist --------------------
+if (!file.exists(gtf_file)) {
+  system(
+    glue::glue("aws s3 cp {gtf_file_uri} {gtf_file} --no-sign-request")
+  )
+}
+if (!file.exists(arm_order_file)) {
+  download.file(url = arm_order_file_url, destfile = arm_order_file)
 }
-  
\ No newline at end of file
+
+# Read and prepare input files -----------------
+
+gtf <- rtracklayer::readGFF(gtf_file)
+
+gene_order_df <- gtf |>
```

Comment:
I'm just suggesting to change all of my pipes to use the magrittr pipe to match your code, so it's all the same here.

```suggestion
gene_order_df <- gtf %>%
```